components/script/dom/range.rs

```diff
@@ -230,36 +252,12 @@ impl Range {
 }
 
 impl RangeMethods for Range {
-    // https://dom.spec.whatwg.org/#dom-range-startcontainer
-    fn StartContainer(&self) -> DomRoot<Node> {
-        self.start.node.get()
-    }
-
-    // https://dom.spec.whatwg.org/#dom-range-startoffset
-    fn StartOffset(&self) -> u32 {
-        self.start.offset.get()
-    }
-
-    // https://dom.spec.whatwg.org/#dom-range-endcontainer
-    fn EndContainer(&self) -> DomRoot<Node> {
-        self.end.node.get()
-    }
-
-    // https://dom.spec.whatwg.org/#dom-range-endoffset
-    fn EndOffset(&self) -> u32 {
-        self.end.offset.get()
-    }
-
-    // https://dom.spec.whatwg.org/#dom-range-collapsed
-    fn Collapsed(&self) -> bool {
-        self.start == self.end
-    }
-
     // https://dom.spec.whatwg.org/#dom-range-commonancestorcontainer
     fn CommonAncestorContainer(&self) -> DomRoot<Node> {
-        let end_container = self.EndContainer();
+        let end_container = self.abstractrange.EndContainer();
```

Comment:
we should be using `.upcast::<AbstractRange>()` here and everywhere else

Reply:
Given that we do this often it would also be acceptible to have an `.abstractrange()` method on `Range` that just upcasts it and returns an `&AbstractRange`

Reply:
Done.